libs/stocks/feature-shell/src/lib/stocks/stocks.component.ts

```diff
@@ -32,12 +26,22 @@ export class StocksComponent implements OnInit {
     });
   }
 
-  ngOnInit() {}
+  ngOnInit() {
```

Comment:
Fixed scoping issue using arrow function.Used debounceTime() to avoid API hits for each keystroke.